finetuner-workflow/finetuner/finetuner.py

```diff
@@ -647,14 +674,17 @@ def __init__(self, path: str, context_length: int = 2048):
         self.length = int(file_stat.st_size / 2 / context_length)
         length_mb = os.stat(path).st_size / (1 << 20)
         num_tokens = self.length * context_length
+        use_uint16 = len(tokenizer) < 0xffff
         self._padding_is_ambiguous = tokenizer.pad_token_id == tokenizer.eos_token_id
-        self._pad_token_id = tokenizer.pad_token_id
+        self._pad_token_id = 0xffff if use_uint16 else 0xffffffff
+        self._token_dtype = numpy.uint16 if use_uint16 else numpy.uint32
         if is_main_process():
             logger.info(f"DATASET: {path}")
             logger.info(
                 f"DATASET SIZE: {length_mb:,.2f}MiB, {num_tokens:,} tokens, "
                 f"{self.length:,} contexts"
             )
+            logger.info(f"DATASET PAD ID: {self._pad_token_id}")
```

Comment:
If the `uint32` functionality isn't working here yet, remove it (or comment it out with an explanation) before merging this PR, so it doesn't confuse people.